crates/bevy_app/src/sub_app.rs

```diff
@@ -429,8 +429,6 @@ impl SubApps {
             sub_app.extract(&mut self.main.world);
             sub_app.update();
         }
-
```

Comment:
Does removing this clear trackers rather than the other one mean that you can't use RemovedComponents in sub apps?

Reply:
RemovedComponents should still work the same for SubApps. clear_trackers is called once here per SubApp (the main app is also a SubApp internally): https://github.com/bevyengine/bevy/pull/13762/files/5888bf276daa1959eeb62671d1a8d6dd230eb6db#diff-1c2be7dcf5607cc6b3a4a9e664e9de0712a5ea313079b5451464513be643e747R136

Reply:
There was some confusion in the past that you couldn't use `Extract<RemovedComponents<T>>` in the extraction subapp. I believe this extra call is here, because the intention was to not clear the trackers at the end of the main app schedule to enable this.

Reply:
Hmmm. If that's the case, then I think we would need another version of `update` For the main app that doesn't call `clear_trackers`.

Reply:
That sounds reasonable. But if it ends up being too involved, it might be better to merge this for 0.14 and fix it in a followup.

Reply:
Wasn't too bad at all once I came up with a test that caught the issue. I couldn't find any problems using RemovedComponents since it has a two frame buffer, but this this was an issue with regular change detection.